### ISSUE TITLE
INTERNAL; CUDA driver error: Requested shared memory size of 819200 larger than allowed size of 166912; while invoking native function hal.executable.create; while calling import; 

### ISSUE THREAD

**Issue by vivekkhandelwal1** [iree-org/iree#12975]
### What happened?

Hi, I'm getting the below error while running an IR module via IREE on cuda:
```
iree-vivekkhandelwal1/runtime/src/iree/hal/drivers/cuda/native_executable.c:133: INTERNAL; CUDA driver error: Requested shared memory size of 819200 larger than allowed size of 166912; while invoking native function hal.executable.create; while calling import; 
[ 1]   native hal.executable.create:0 -
[ 0] bytecode module.__init:238 cuda_failure.mlir:9:10
      at cuda_failure.mlir:4:3
```

### Steps to reproduce your issue

In order to reproduce the issue, run the following commands over this IR:
```
#map = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
module attributes {torch.debug_module_name = "GraphModule"} {
  ml_program.global private mutable @global_seed(dense<0> : tensor<i64>) : tensor<i64>
  func.func @forward(%arg0: tensor<320x640x1x1xf32>) -> tensor<640x320x1x1xf32> {
    %cst = arith.constant 0.000000e+00 : f32
    %c0 = arith.constant 0 : index
    %0 = tensor.empty() : tensor<640x320x1x1xf32>
    %1 = linalg.fill ins(%cst : f32) outs(%0 : tensor<640x320x1x1xf32>) -> tensor<640x320x1x1xf32>
    %2 = linalg.generic {indexing_maps = [#map], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} outs(%1 : tensor<640x320x1x1xf32>) {
    ^bb0(%out: f32):
      %3 = linalg.index 0 : index
      %4 = linalg.index 1 : index
      %5 = linalg.index 2 : index
      %6 = linalg.index 3 : index
      %7 = arith.subi %c0, %5 : index
      %8 = arith.subi %c0, %6 : index
      %extracted = tensor.extract %arg0[%4, %3, %7, %8] : tensor<320x640x1x1xf32>
      linalg.yield %extracted : f32
    } -> tensor<640x320x1x1xf32>
    return %2 : tensor<640x320x1x1xf32>
  }
}
```
Run:
1.)
```
./build/tools/iree-compile --iree-input-type=tm_tensor --iree-vm-bytecode-module-output-format=flatbuffer-binary --iree-hal-target-backends=cuda --mlir-print-debuginfo --mlir-print-op-on-diagnostic=false --iree-stream-resource-index-bits=64 --iree-vm-target-index-bits=64 --iree-util-zero-fill-elided-attrs sd_linalg_ir_crash_cuda_07_04.mlir -o cuda_failure.vmfb
```
2.)
```
./build/tools/iree-run-module --module=cuda_failure.vmfb --device=cuda --function=forward --input=320x640x1x1xf32
```
With this, you will get the above-mentioned error.

### What component(s) does this issue relate to?

MLIR, Runtime

### Version information

Git commit hash: `366edbfd7ea1116b7465224ccd7ae61a470bb78e`

**Comment by vivekkhandelwal1** [comment]
Hi @ThomasRaoux, could you please take a look at this issue?

**Comment by ThomasRaoux** [comment]
Fix under review:
https://github.com/openxla/iree/pull/12977

**Comment by vivekkhandelwal1** [comment]
> Fix under review: #12977

Thanks for fixing this!
